fraisier sync: handle existing PR + stale local branch gracefully

## Problem

`fraisier sync staging` and `fraisier sync main` (invoked via `make pr-sync-staging` / `make pr-sync-main` wrappers in a downstream project) have two fragility modes that bite when re-running after a previous attempt was interrupted or completed in a non-standard way:

### 1. Stale local sync branch from previous run

If a prior `fraisier sync main` run was interrupted after creating `sync/main-from-staging` locally but before pushing/cleanup, the next invocation hits:

```
Creating sync/main-from-staging from origin/staging
fatal: a branch named 'sync/main-from-staging' already exists
Error: Command '['git', 'checkout', '-b', 'sync/main-from-staging', 'origin/staging']'
       returned non-zero exit status 128.
make: *** [Makefile:690: pr-sync-main] Error 1
```

Workaround: `git branch -D sync/main-from-staging` then retry. Annoying because the natural reaction is to retry the command, not to inspect local branch state.

### 2. Existing remote PR for the same head branch

If the sync was previously attempted and a PR was created — even if the previous push hash differs from the current one — `gh pr create` fails. The fraisier output swallows the gh error and reports a generic `subprocess.CalledProcessError`:

```
Pushing sync/staging-from-dev
To github.com:org/repo.git
   abc1234..def5678  sync/staging-from-dev -> sync/staging-from-dev
Creating PR: sync/staging-from-dev → staging
Error: Command '['gh', 'pr', 'create', ...]' returned non-zero exit status 1.
```

Behind the scenes, gh prints something like:

```
a pull request for branch \"sync/staging-from-dev\" into branch \"staging\" already exists:
https://github.com/org/repo/pull/1234
```

The branch *was* updated (the push succeeded), but the user has no idea unless they go look — and the sync command exited non-zero, suggesting it should be retried. Retrying just hits the same error.

## Proposed handling

Both of these are recoverable conditions that fraisier already has enough information to handle:

### For the stale local branch

Before `git checkout -b <sync-branch> <upstream>`, check if `<sync-branch>` exists locally:

```python
if local_branch_exists(sync_branch):
    if branch_has_unpushed_changes(sync_branch):
        # Loud — the user might lose work
        bail(f\"{sync_branch} exists locally with unpushed commits; delete or push manually\")
    else:
        run([\"git\", \"branch\", \"-D\", sync_branch])     # safe to delete; it'll be recreated
```

Alternatively: just use `git checkout -B` (force-create) instead of `git checkout -b`. That's a one-character fix and matches the intent: \"I want this branch pointed at upstream/X, regardless of any local state\".

### For the existing remote PR

After `git push` succeeds, check whether a PR already exists for `<sync-branch> → <target>`:

```python
existing = gh_pr_view(head=sync_branch, base=target_branch)
if existing:
    log(f\"Existing PR {existing.url} updated (head: {sync_branch}@{new_sha})\")
    if auto_merge:
        gh_pr_merge(existing.number, '--auto', '--' + merge_method)
    return success(existing.url)
else:
    gh_pr_create(...)
```

Idempotent on retries. Same observable behavior whether the PR was just created or already existed.

## Why this matters

`fraisier sync` is the canonical way to promote dev → staging → main in projects following the fraisier convention. When it fails on these recoverable conditions, the failure mode is opaque: the user sees a stack trace, doesn't know whether the underlying push/merge succeeded, and has to manually inspect git state + GitHub.

The fix is small (handful of lines per case) and the surface is well-scoped (`fraisier/cli/sync.py` or wherever the sync orchestration lives).

## Related

- fraiseql/fraisier#204 — post-migration verification hooks (different bucket of robustness work on the deploy lifecycle)
- fraiseql/confiture#120 — ACL-aware drift (also defensive, also adjacent)

## Definition of Done

- [ ] `fraisier sync` succeeds idempotently when re-invoked after a previous attempt left a stale local sync branch.
- [ ] `fraisier sync` succeeds idempotently when an existing open PR matches the same head→base — by updating the PR rather than failing.
- [ ] Error output mentions the existing PR URL when the conditions overlap (so the user knows where to look).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,19 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.21.1] - 2026-05-22
+
+### Fixed
+
+- **`fraisier sync` is now idempotent on retry** ([#213](https://github.com/fraiseql/fraisier/issues/213)). Re-running after a previous interrupted or completed sync no longer fails:
+  - The local sync branch is force-created (`git checkout -B`), so a stale branch from an interrupted prior run is silently overwritten by the fresh checkout. Sync branches are fraisier-owned end-to-end; any commits on them are re-derived from `origin/<source>` + pre-merge on every run.
+  - After pushing, fraisier checks for an existing PR for the sync branch via `gh pr view`. When the prior PR is OPEN, it re-enables auto-merge (idempotent for `--squash`) on the existing PR instead of failing on `gh pr create`. The PR URL appears in the success message as `PR updated and auto-merge enabled: <url>`.
+  - When the prior PR is CLOSED or MERGED, fraisier opens a fresh PR (GitHub allows reusing a head ref after closure). The prior PR URL is logged for operator context.
+
+### Upgrade notes
+
+- A user who has manually committed to a sync branch will lose those commits on the next `fraisier sync` invocation (the new `-B` overwrites the local branch). Sync branches were already documented as fraisier-owned; if you want to keep commits on a sync branch, push them to a non-sync ref.
+
 ## [0.21.0] - 2026-05-22
 
 ### Added

--- a/fraisier/cli/sync.py
+++ b/fraisier/cli/sync.py
@@ -127,6 +127,28 @@ def _capture(cmd: list[str]) -> subprocess.CompletedProcess:
     return subprocess.run(cmd, check=True, capture_output=True, text=True)
 
 
+def _find_existing_pr(sync_branch: str) -> dict | None:
+    """Return PR head-ref lookup result, or None if no PR exists for the branch.
+
+    Result shape: ``{"url": str, "state": "OPEN" | "CLOSED" | "MERGED"}``.
+    Uses ``check=False`` because ``gh`` exits non-zero when no PR matches
+    the head ref — a normal "nothing here yet" signal, not an error.
+    Drafts are reported as state OPEN (the ``isDraft`` bit lives in a
+    separate field we don't query); re-enabling auto-merge on a draft is
+    the desired behavior — GitHub queues the merge for when the PR is
+    marked ready.
+    """
+    result = subprocess.run(
+        ["gh", "pr", "view", sync_branch, "--json", "url,state"],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    if result.returncode != 0:
+        return None
+    return json.loads(result.stdout)
+
+
 def _commit_if_staged(message: str) -> None:
     """Commit the staged index with ``message`` only if there is something staged.
 
@@ -359,6 +381,24 @@ def sync_cmd(
 
         console.print(f"  Pushing [bold]{sync_branch}[/bold]")
         _run(["git", "push", "origin", sync_branch])
+
+        existing = _find_existing_pr(sync_branch)
+        if existing and existing["state"] == "OPEN":
+            pr_url = existing["url"]
+            console.print(
+                f"  Existing open PR found, updating: [bold]{pr_url}[/bold]"
+            )
+            _run(["gh", "pr", "merge", "--auto", "--squash", pr_url])
+            subprocess.run(["git", "checkout", original_branch], check=False)
+            console.print(
+                f"==> [green]Done.[/green] PR updated and auto-merge enabled: {pr_url}"
+            )
+            return
+        if existing:
+            console.print(
+                f"  Prior PR for {sync_branch} was {existing['state'].lower()}: "
+                f"{existing['url']} — opening a new one"
+            )
 
         console.print(f"  Creating PR: [bold]{sync_branch}[/bold] → [bold]{tgt}[/bold]")
         pr_result = subprocess.run(

--- a/fraisier/cli/sync.py
+++ b/fraisier/cli/sync.py
@@ -385,9 +385,7 @@ def sync_cmd(
         existing = _find_existing_pr(sync_branch)
         if existing and existing["state"] == "OPEN":
             pr_url = existing["url"]
-            console.print(
-                f"  Existing open PR found, updating: [bold]{pr_url}[/bold]"
-            )
+            console.print(f"  Existing open PR found, updating: [bold]{pr_url}[/bold]")
             _run(["gh", "pr", "merge", "--auto", "--squash", pr_url])
             subprocess.run(["git", "checkout", original_branch], check=False)
             console.print(

--- a/fraisier/cli/sync.py
+++ b/fraisier/cli/sync.py
@@ -149,7 +149,7 @@ def _print_dry_run_plan(source: str, tgt: str, sync_branch: str) -> None:
     console.print()
     console.print("  Would run:")
     console.print(f"    git fetch origin {source} {tgt}")
-    console.print(f"    git checkout -b {sync_branch} origin/{source}")
+    console.print(f"    git checkout -B {sync_branch} origin/{source}")
     console.print(f"    git merge origin/{tgt} --no-edit --no-commit")
     console.print(
         f"    # conflicts in [{auto_owned}] auto-resolved from {source};"
@@ -276,7 +276,7 @@ def sync_cmd(
     branch_created = False
     try:
         console.print(f"  Creating [bold]{sync_branch}[/bold] from origin/{source}")
-        _run(["git", "checkout", "-b", sync_branch, f"origin/{source}"])
+        _run(["git", "checkout", "-B", sync_branch, f"origin/{source}"])
         branch_created = True
 
         console.print(f"  Pre-merging origin/{tgt} into {sync_branch}")

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,7 +31,7 @@ license = {text = "MIT"}
 name = "fraisier"
 readme = "README.md"
 requires-python = ">=3.11"
-version = "0.21.0"
+version = "0.21.1"
 
 [project.optional-dependencies]
 all-databases = [

--- a/tests/test_sync.py
+++ b/tests/test_sync.py
@@ -1002,7 +1002,10 @@ class TestSyncBranchForceCreate:
         checkout_create = [
             cmd
             for cmd in commands
-            if len(cmd) >= 2 and cmd[0] == "git" and cmd[1] == "checkout" and "-B" in cmd
+            if len(cmd) >= 2
+            and cmd[0] == "git"
+            and cmd[1] == "checkout"
+            and "-B" in cmd
         ]
         assert checkout_create, (
             "expected a `git checkout -B sync/...` call; got: "
@@ -1092,9 +1095,9 @@ class TestSyncExistingPR:
         assert result.exit_code == 0, result.output
         commands = [c[0][0] for c in m.call_args_list]
         # gh pr create must NOT be called for an existing OPEN PR
-        assert not any(
-            cmd[:3] == ["gh", "pr", "create"] for cmd in commands
-        ), f"gh pr create was called; commands: {commands}"
+        assert not any(cmd[:3] == ["gh", "pr", "create"] for cmd in commands), (
+            f"gh pr create was called; commands: {commands}"
+        )
         # gh pr merge --auto --squash <existing-url> IS called
         assert [
             "gh",
@@ -1122,7 +1125,9 @@ class TestSyncExistingPR:
         commands = [c[0][0] for c in m.call_args_list]
         # gh pr create IS called for a CLOSED prior PR
         create_calls = [cmd for cmd in commands if cmd[:3] == ["gh", "pr", "create"]]
-        assert len(create_calls) == 1, f"expected 1 gh pr create call; got {create_calls}"
+        assert len(create_calls) == 1, (
+            f"expected 1 gh pr create call; got {create_calls}"
+        )
         # New PR URL is the one in the success message
         assert self.NEW_PR_URL in result.output
         # Prior PR URL is surfaced as informational context

--- a/tests/test_sync.py
+++ b/tests/test_sync.py
@@ -906,7 +906,7 @@ class TestSyncDryRun:
         cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
         with patch(_PATCH):
             result = CliRunner().invoke(main, ["-c", cfg, "sync", "--dry-run"])
-        assert "git checkout -b sync/staging-from-dev origin/dev" in result.output
+        assert "git checkout -B sync/staging-from-dev origin/dev" in result.output
 
     def test_dry_run_shows_merge(self, tmp_path):
         cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
@@ -948,3 +948,67 @@ class TestSyncDryRun:
         assert "staging" in result.output
         assert "prod" in result.output
         assert "git fetch origin staging prod" in result.output
+
+
+# ---------------------------------------------------------------------------
+# CLI: branch force-create (issue #213, fragility 1)
+# ---------------------------------------------------------------------------
+
+
+class TestSyncBranchForceCreate:
+    """Sync branch is force-created (-B) so a stale local branch from an
+    interrupted prior run is silently overwritten on re-invocation."""
+
+    SHA = "deadbeef" * 5
+    MERGE_BASE = "cafe1234" * 5
+    PR_URL = "https://github.com/org/repo/pull/42"
+
+    def _side_effects(self):
+        return [
+            _mk(stdout="main\n"),  # git rev-parse HEAD
+            _mk(),  # git fetch
+            _mk(stdout=self.SHA + "\n"),  # git rev-parse origin/dev
+            _mk(stdout=self.MERGE_BASE + "\n"),  # git merge-base
+            _mk(returncode=0, stdout='{"version": "1.1.0"}'),  # version dev
+            _mk(returncode=0, stdout='{"version": "1.0.0"}'),  # version staging
+            _mk(),  # git checkout -B
+            _mk(),  # git merge (clean)
+            _mk(returncode=1),  # git diff --cached (staged)
+            _mk(),  # git commit pre-merge
+            _mk(),  # git push
+            _mk(stdout=self.PR_URL + "\n"),  # gh pr create
+            _mk(),  # gh pr merge
+            _mk(),  # git checkout main
+        ]
+
+    def test_uses_capital_B_to_force_create(self, tmp_path):
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = self._side_effects()
+            CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        commands = [c[0][0] for c in m.call_args_list]
+        checkout_create = [
+            cmd
+            for cmd in commands
+            if len(cmd) >= 2 and cmd[0] == "git" and cmd[1] == "checkout" and "-B" in cmd
+        ]
+        assert checkout_create, (
+            "expected a `git checkout -B sync/...` call; got: "
+            f"{[c for c in commands if 'checkout' in c]}"
+        )
+
+    def test_does_not_use_lowercase_b(self, tmp_path):
+        """Regression guard: lowercase -b fails on a stale local branch."""
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = self._side_effects()
+            CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        commands = [c[0][0] for c in m.call_args_list]
+        lowercase = [
+            cmd
+            for cmd in commands
+            if len(cmd) >= 3 and cmd[:2] == ["git", "checkout"] and "-b" in cmd
+        ]
+        assert not lowercase, (
+            f"sync must not use `git checkout -b` (use -B); offenders: {lowercase}"
+        )

--- a/tests/test_sync.py
+++ b/tests/test_sync.py
@@ -378,6 +378,7 @@ class TestSyncHappyPath:
             _mk(returncode=1),  # git diff --cached (staged)
             _mk(),  # git commit pre-merge
             _mk(),  # git push
+            _mk(returncode=1),  # gh pr view (no existing PR)
             _mk(stdout=self.PR_URL + "\n"),  # gh pr create
             _mk(),  # gh pr merge
             _mk(),  # git checkout main
@@ -430,6 +431,7 @@ class TestSyncHappyPath:
                 _mk(),  # git merge (clean)
                 _mk(returncode=0),  # git diff --cached --quiet (nothing staged)
                 _mk(),  # git push (no commit)
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),
                 _mk(),
                 _mk(),  # git checkout main
@@ -483,6 +485,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -512,6 +515,7 @@ class TestSyncConflicts:
                 _mk(returncode=1),  # git diff --cached --quiet → something staged
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -548,6 +552,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(returncode=0),  # git diff --cached --quiet → nothing staged
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -587,6 +592,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -674,6 +680,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -708,6 +715,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -743,6 +751,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -776,6 +785,7 @@ class TestSyncConflicts:
                 _mk(stdout=""),  # diff --filter=U (remaining: clean)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=self.PR_URL + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -828,6 +838,7 @@ class TestSyncConfirmation:
                 _mk(returncode=1),  # git diff --cached (staged)
                 _mk(),  # git commit
                 _mk(),  # git push
+                _mk(returncode=1),  # gh pr view (no existing PR)
                 _mk(stdout=pr_url + "\n"),  # gh pr create
                 _mk(),  # gh pr merge
                 _mk(),  # git checkout main
@@ -976,6 +987,7 @@ class TestSyncBranchForceCreate:
             _mk(returncode=1),  # git diff --cached (staged)
             _mk(),  # git commit pre-merge
             _mk(),  # git push
+            _mk(returncode=1),  # gh pr view (no existing PR)
             _mk(stdout=self.PR_URL + "\n"),  # gh pr create
             _mk(),  # gh pr merge
             _mk(),  # git checkout main
@@ -1011,4 +1023,167 @@ class TestSyncBranchForceCreate:
         ]
         assert not lowercase, (
             f"sync must not use `git checkout -b` (use -B); offenders: {lowercase}"
+        )
+
+
+# ---------------------------------------------------------------------------
+# CLI: existing-PR detection (issue #213, fragility 2)
+# ---------------------------------------------------------------------------
+
+
+class TestSyncExistingPR:
+    """After pushing, sync looks up an existing PR for the sync branch via
+    `gh pr view`. OPEN → update + re-enable auto-merge + exit; CLOSED or
+    MERGED → log the prior URL and open a fresh PR; absent → unchanged
+    behavior (open a new PR)."""
+
+    SHA = "deadbeef" * 5
+    MERGE_BASE = "cafe1234" * 5
+    EXISTING_PR_URL = "https://github.com/org/repo/pull/40"
+    NEW_PR_URL = "https://github.com/org/repo/pull/42"
+
+    def _calls_up_to_push(self):
+        """Mocked subprocess calls from sync start through `git push`."""
+        return [
+            _mk(stdout="main\n"),  # git rev-parse HEAD
+            _mk(),  # git fetch
+            _mk(stdout=self.SHA + "\n"),  # git rev-parse origin/dev
+            _mk(stdout=self.MERGE_BASE + "\n"),  # git merge-base
+            _mk(returncode=0, stdout='{"version": "1.1.0"}'),  # version dev
+            _mk(returncode=0, stdout='{"version": "1.0.0"}'),  # version staging
+            _mk(),  # git checkout -B
+            _mk(),  # git merge (clean)
+            _mk(returncode=1),  # git diff --cached (staged)
+            _mk(),  # git commit pre-merge
+            _mk(),  # git push
+        ]
+
+    def _pr_view_open(self):
+        return _mk(
+            returncode=0,
+            stdout=f'{{"url":"{self.EXISTING_PR_URL}","state":"OPEN"}}',
+        )
+
+    def _pr_view_closed(self):
+        return _mk(
+            returncode=0,
+            stdout=f'{{"url":"{self.EXISTING_PR_URL}","state":"CLOSED"}}',
+        )
+
+    def _pr_view_merged(self):
+        return _mk(
+            returncode=0,
+            stdout=f'{{"url":"{self.EXISTING_PR_URL}","state":"MERGED"}}',
+        )
+
+    def _pr_view_absent(self):
+        return _mk(returncode=1, stderr="no pull requests found for branch")
+
+    def test_updates_existing_open_pr_instead_of_creating(self, tmp_path):
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = [
+                *self._calls_up_to_push(),
+                self._pr_view_open(),
+                _mk(),  # gh pr merge --auto --squash
+                _mk(),  # git checkout main
+            ]
+            result = CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        assert result.exit_code == 0, result.output
+        commands = [c[0][0] for c in m.call_args_list]
+        # gh pr create must NOT be called for an existing OPEN PR
+        assert not any(
+            cmd[:3] == ["gh", "pr", "create"] for cmd in commands
+        ), f"gh pr create was called; commands: {commands}"
+        # gh pr merge --auto --squash <existing-url> IS called
+        assert [
+            "gh",
+            "pr",
+            "merge",
+            "--auto",
+            "--squash",
+            self.EXISTING_PR_URL,
+        ] in commands
+        assert "updated" in result.output
+        assert self.EXISTING_PR_URL in result.output
+
+    def test_creates_new_pr_when_existing_is_closed(self, tmp_path):
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = [
+                *self._calls_up_to_push(),
+                self._pr_view_closed(),
+                _mk(stdout=self.NEW_PR_URL + "\n"),  # gh pr create
+                _mk(),  # gh pr merge
+                _mk(),  # git checkout main
+            ]
+            result = CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        assert result.exit_code == 0, result.output
+        commands = [c[0][0] for c in m.call_args_list]
+        # gh pr create IS called for a CLOSED prior PR
+        create_calls = [cmd for cmd in commands if cmd[:3] == ["gh", "pr", "create"]]
+        assert len(create_calls) == 1, f"expected 1 gh pr create call; got {create_calls}"
+        # New PR URL is the one in the success message
+        assert self.NEW_PR_URL in result.output
+        # Prior PR URL is surfaced as informational context
+        assert self.EXISTING_PR_URL in result.output
+        assert "closed" in result.output.lower()
+
+    def test_creates_new_pr_when_existing_is_merged(self, tmp_path):
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = [
+                *self._calls_up_to_push(),
+                self._pr_view_merged(),
+                _mk(stdout=self.NEW_PR_URL + "\n"),  # gh pr create
+                _mk(),  # gh pr merge
+                _mk(),  # git checkout main
+            ]
+            result = CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        assert result.exit_code == 0, result.output
+        commands = [c[0][0] for c in m.call_args_list]
+        assert any(cmd[:3] == ["gh", "pr", "create"] for cmd in commands)
+        assert self.NEW_PR_URL in result.output
+        assert self.EXISTING_PR_URL in result.output
+        assert "merged" in result.output.lower()
+
+    def test_creates_new_pr_when_no_existing_pr(self, tmp_path):
+        """No PR exists for the sync branch → behavior unchanged: create one."""
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = [
+                *self._calls_up_to_push(),
+                self._pr_view_absent(),
+                _mk(stdout=self.NEW_PR_URL + "\n"),  # gh pr create
+                _mk(),  # gh pr merge
+                _mk(),  # git checkout main
+            ]
+            result = CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        assert result.exit_code == 0, result.output
+        commands = [c[0][0] for c in m.call_args_list]
+        assert any(cmd[:3] == ["gh", "pr", "create"] for cmd in commands)
+        assert self.NEW_PR_URL in result.output
+        # No prior PR mentioned
+        assert self.EXISTING_PR_URL not in result.output
+
+    def test_existing_open_pr_url_on_final_done_line(self, tmp_path):
+        """Operators grep the final `==> Done.` line for the PR URL — it
+        must be on the same line as 'Done.' even on the retry path."""
+        cfg = _setup(tmp_path, [{"source": "dev", "target": "staging"}])
+        with patch(_PATCH) as m:
+            m.side_effect = [
+                *self._calls_up_to_push(),
+                self._pr_view_open(),
+                _mk(),  # gh pr merge
+                _mk(),  # git checkout main
+            ]
+            result = CliRunner().invoke(main, ["-c", cfg, "sync", "--yes"])
+        done_lines = [
+            line
+            for line in result.output.splitlines()
+            if "Done." in line and self.EXISTING_PR_URL in line
+        ]
+        assert done_lines, (
+            "expected the existing PR URL on the final `==> Done.` line; "
+            f"output:\n{result.output}"
         )

